python_modules/libraries/dagster-azure/dagster_azure/blob/blob_fake_resource.py

```diff
@@ -0,0 +1,164 @@
+from collections import defaultdict
+from contextlib import contextmanager
+import io
+import random
+
+from azure.core.exceptions import ResourceNotFoundError
+
+from dagster.seven import mock
+
+
+def create_blob_fake_resource(account_name):
```

Comment:
I see that you're copying from `s3_fake_resource`, but I think it's a little confusing to use the term "resource" here, because there's no Dagster resource definition involved.  I also think it's slightly weird to have the `Fake` in the middle of the name.  Let me know what you think of these suggestions:
* Eliminate the `create_blob_fake_resource` function and make `fake-creds` the default value for the `BlobFakeClient` constructor.
* Rename the  `BlobFakeClient` to `FakeBlobClient`
* Rename blob_fake_resource.py to fake_blob_client.py

Reply:
Or actually, reading a little more, would these make sense?
* `BlobFakeClient` -> `FakeBlobServiceClient`
* `BlobFakeContainerClient` -> `FakeBlobContainerClient`
* `BlobFakeBlobClient` -> `FakeBlobClient`

Reply:
Yep that sounds a lot nicer to me. I was going for consistency with S3 but I much prefer starting with `Fake`!